src/comments.ts

```diff
@@ -0,0 +1,297 @@
+        document: vscode.TextDocument,
+        token: vscode.CancellationToken
+    ): Promise<vscode.CommentInfo> {
+        let threads: vscode.CommentThread[] = []
+        try {
+            threads = await provideDocumentComments(document)
+        } catch (e) {
+            log.appendLine(`provideDocumentComments ${e}`)
+        }
+
+        const lastLine = document.lineCount - 1
+        const commentingRanges = [new vscode.Range(0, 0, lastLine, document.lineAt(lastLine).range.end.character)]
+
+        return { threads, commentingRanges }
+    }
+
+    public async createNewCommentThread(
+        document: vscode.TextDocument,
+        range: vscode.Range,
+        text: string,
+        token: vscode.CancellationToken
+    ): Promise<vscode.CommentThread> {
+        return await createNewCommentThread(document, range, text)
+    }
+
+    public async replyToCommentThread(
+        document: vscode.TextDocument,
+        range: vscode.Range,
+        commentThread: vscode.CommentThread,
+        text: string,
+        token: vscode.CancellationToken
+    ): Promise<vscode.CommentThread> {
+        return await replyToCommentThread(document, range, commentThread, text)
+    }
+
+    private didChangeCommentThreads = new vscode.EventEmitter<vscode.CommentThreadChangedEvent>()
+
+    public onDidChangeCommentThreads = this.didChangeCommentThreads.event
+}
+
+async function provideDocumentComments(document: vscode.TextDocument): Promise<vscode.CommentThread[]> {
+    log.appendLine(`provideDocumentComments ${document.uri}`)
+    const [remoteUrl, branch, path] = await repoInfo(document.fileName)
+    if (remoteUrl === '') {
+        throw new Error('Git repository has no remote url configured')
+    }
+    const data = await queryGraphQL(
+        gql`
+            query DiscussionThreads(
+                $targetRepositoryGitCloneURL: String!
+                $targetRepositoryPath: String!
+                $relativeRev: String!
+            ) {
+                discussionThreads(
+                    first: 10000
+                    targetRepositoryGitCloneURL: $targetRepositoryGitCloneURL
+                    targetRepositoryPath: $targetRepositoryPath
+                ) {
+                    totalCount
+                    pageInfo {
+                        hasNextPage
+                    }
+                    nodes {
+                        ...DiscussionThreadFields
+                    }
+                }
+            }
+            ${discussionThreadFieldsFragment}
+        `,
+        {
+            targetRepositoryGitCloneURL: remoteUrl,
+            targetRepositoryPath: path,
+            relativeRev: branch,
+        }
+    )
+
+    if (!data.discussionThreads || !data.discussionThreads.nodes) {
+        throw new Error(`Invalid GraphQL response for DiscussionThreads`)
+    }
+
+    const threads: vscode.CommentThread[] = []
+    for (const thread of data.discussionThreads.nodes) {
```

Comment:
You need to ignore threads whose `thread.target.__typename !== 'DiscussionThreadTargetRepo'` or else this will break when I add more in the future.